### ISSUE TITLE
Unexpected focus/scroll when error present

### ISSUE THREAD

**Issue by marksreeves** [BorderTech/wcomponents#1253]
If 
1. a WMessages or WMessageBox is used to display a page level error; and
2. the error is not focusable (so not generated from WValidationErrors); and
3. the page loads with a default focus set (e.g. from WButton auto-refocus as per #501 )
then the button will receive focus and the page may scroll to hide the messages.

We should check for error boxes before doing autofocus. We _could_ check for all boxed messages.

@ricksbrown I think it should be enough to not honour focus request (on page load - not from an Ajax response) if there is an **ERROR** level message box. We could test document position to try to determine if the element to be focussed is above the message box but I think this is overkill. If there are new error messages we should assume a change of context and not set focus within the page content. Thoughts?
